src/cmd/watch.rs

```diff
@@ -179,3 +187,50 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ignore::gitignore::GitignoreBuilder;
+    use std::env;
+
+    fn path_to_buf(root: &str, file_name: &str) -> PathBuf {
+        let mut path = PathBuf::from(root);
+        path.push(file_name);
+        path
+    }
+
+    #[test]
+    fn test_filter_ignored_files() {
+        let current_dir = env::current_dir().unwrap();
+
+        let ignore = GitignoreBuilder::new(&current_dir)
+            .add_line(None, "*.html")
+            .unwrap()
+            .build()
+            .unwrap();
+        let should_remain = path_to_buf(current_dir.to_str().unwrap(), "record.text");
+        let should_filter = path_to_buf(current_dir.to_str().unwrap(), "index.html");
+
+        let remain = filter_ignored_files(ignore, &[should_remain.clone(), should_filter]);
+        assert_eq!(remain, vec![should_remain])
+    }
+
+    #[test]
+    fn filter_ignored_files_should_handle_parent_dir() {
+        let current_dir = env::current_dir().unwrap();
+
+        let ignore = GitignoreBuilder::new(&current_dir)
+            .add_line(None, "*.html")
+            .unwrap()
+            .build()
+            .unwrap();
+
+        let parent_dir = format!("{}/../", current_dir.to_str().unwrap());
```

Comment:
Similarly, this can be:

```suggestion
        let parent_dir = current_dir.join("..");
```

Reply:
Fix 40c034e.